### PR TITLE
Add pool size parameter to client constructor

### PR DESCRIPTION
**Pull request by vaniakov**
This pull request relates to the issue [Connection pool is full, discarding connection](https://github.com/influxdata/influxdb-python/issues/349). To reproduce it you need to create more than 10 (default size of the urllib3 connection pool) simultaneous connection to the database from the same client object.

### CODE REVIEW

File: influxdb/client.py [hunk 1]
```diff
@@ -64,6 +66,7 @@ def __init__(self,
                  port=8086,
                  username='root',
                  password='root',
+                 pool_size=10,
```
Comment (xginn8) [inline influxdb/client.py]: I think it makes sense to cast this value to an int, like we do for the `port` parameter in line 81.

Comment (vaniakov) [reply influxdb/client.py]: @xginn8 make sense.

**Comment by vaniakov** [comment]
@xginn8 nice catch!

**Comment by xginn8** [comment]
thanks for contributing!

**Comment by hppylbb** [comment]
Hi all, sorry but I find this issue is still exist in the master branch, the parameter pool_size is invalid even we added it to \_\_init\_\_ method. I think that https://github.com/influxdata/influxdb-python/blob/35732cd7dfe5a585564999c9f881bd88e7c1531d/influxdb/client.py#L106 should be modified to "`self._session.mount(self._scheme+'://', adapter)`",
because after executing this line, the "session.adapters" will have three members: "`OrderedDict([('https://', <requests.adapters.HTTPAdapter object at 0x7f0c9c123890>), ('http://', <requests.adapters.HTTPAdapter object at 0x7f0c9404e190>), ('http', <requests.adapters.HTTPAdapter object at 0x7f0c9e874290>)]))`", the https:// and http:// adapters are added here https://github.com/requests/requests/blob/3dc84cde5b785a85cb20eb2f0d4530fd568a6af7/requests/sessions.py#L391-L393 while initializing, and when you execute https://github.com/requests/requests/blob/3dc84cde5b785a85cb20eb2f0d4530fd568a6af7/requests/sessions.py#L691, you want to get the adapter 'http' which is mounted at https://github.com/influxdata/influxdb-python/blob/35732cd7dfe5a585564999c9f881bd88e7c1531d/influxdb/client.py#L106 (the third one in OrderedDict),  but here you got a incorrect adapter which is http:// (the second one in OrderedDict), this adapter doesn't contain the 'pool_size' parameter we added to \_\_init\_\_ method while initializing InfluxDBClient.

So while we write_points to influxdb, we still use the default pool_size=10, then we run into issue https://github.com/influxdata/influxdb-python/issues/349.

If we add '://' to `self._session.mount(self._scheme+'://', adapter)`, then the OrderedDict will only contains two adapters: `OrderedDict([('https://', <requests.adapters.HTTPAdapter object at 0x7f0c9c123890>), ('http://', <requests.adapters.HTTPAdapter object at 0x7f0c9404e190>)])`, the http:// adapter will be overrided by our session mount method, then we can get the correct http adapter with the pool_size we added.

**Comment by vaniakov** [comment]
@hppylbb you are correct. Working on the new PR.

**Pull request merged**